File: tooling/archetypes/camel-archetype-component/src/main/resources/archetype-resources/src/main/java/__name__Endpoint.java
package ${package};

import org.apache.camel.Consumer;
import org.apache.camel.Processor;
import org.apache.camel.Producer;
import org.apache.camel.impl.DefaultEndpoint;

/**
 * Represents a ${name} endpoint.
 */
public class ${name}Endpoint extends DefaultEndpoint {

    public ${name}Endpoint() {
    }

    public ${name}Endpoint(String uri, ${name}Component component) {
        super(uri, component);
    }

    public ${name}Endpoint(String endpointUri) {
        super(endpointUri);
    }

    public Producer createProducer() throws Exception {
        return new ${name}Producer(this);
    }

    public Consumer createConsumer(Processor processor) throws Exception {
        return new ${name}Consumer(this, processor);
    }

    public boolean isSingleton() {
        return true;
    }
}

File: tooling/archetypes/camel-archetype-component-scala/src/main/resources/archetype-resources/src/main/scala/__name__Endpoint.scala
package ${package}

import org.apache.camel.{Consumer, Processor, Producer}
import org.apache.camel.impl.DefaultEndpoint

/**
 * Represents a ${name} endpoint.
 */
class ${name}Endpoint(uri:String, component: ${name}Component) extends DefaultEndpoint(uri, component) {

  def createProducer() : Producer = {
    new ${name}Producer(this)
  }

  def createConsumer(processor: Processor) : Consumer = {
    new ${name}Consumer(this, processor)
  }

  def isSingleton() : Boolean = true

}
